### PR TITLE
Promote JobPodFailurePolicy and PodDisruptionConditions e2e tests to Conformance

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind documentation

#### Which issue(s) this PR fixes:

Tracking issue: https://github.com/kubernetes/enhancements/issues/3329

#### Special notes for your reviewer:

Actual commit for the conformance tests is the second. The first commit marks the feature gates as GA, so the conformance suite only runs GA tests. The first commit does not lock the feature gates, so that the diff is minimal. 
When the two PRs are merged, then this commit can be removed entirely:
- https://github.com/kubernetes/kubernetes/pull/125461
- https://github.com/kubernetes/kubernetes/pull/125442

Promoting tests to conformance instructions: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#promoting-tests-to-conformance 

There are no failures according to the testgrids:
Testgrid https://testgrid.k8s.io/sig-apps#gce (103 builds)
- "Job should allow to use the pod failure policy on exit code to fail the job early"
- "Job should allow to use the pod failure policy to not count the failure towards the backoffLimit"
- "Job should allow to use a pod failure policy to ignore failure for an evicted pod; matching on the exit code"
- "should allow to use a pod failure policy to ignore failure for an evicted pod; matching on the DisruptionTarget condition"

Testgrid https://testgrid.k8s.io/sig-scheduling#gce-serial (76 builds)
- "validates pod disruption condition is added to the preempted pod"

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
https://github.com/kubernetes/enhancements/tree/master/keps/sig-apps/3329-retriable-and-non-retriable-failures
```

### CODE REVIEW

**Comment by mimowo** [comment]
/remove-area kubelet
/remove-sig api-machinery
/remove-sig node
These don't seem necessary if I don't lock the FG in this PR, to minimize the diff.

File: test/conformance/testdata/conformance.yaml [hunk 1]
```diff
@@ -1055,6 +1055,60 @@
...
+  codename: '[sig-apps] Job should allow to use a pod failure policy to ignore failure
+    for an evicted pod; matching on the DisruptionTarget condition [Conformance]'
+  description: 'This test is using an indexed job. The pod corresponding to the 0th
+    index creates a marker file on the host and runs ''forever'' until evicted. We
+    use the non-0-indexed pods to determine if the marker file is already created
+    by the 0th indexed pod - the non-0-indexed pods fail and restart until the marker
+    file is created (their potential failures are ignored based on the exit code).
+    Once the marker file is created the 0th indexed pod is evicted (DisruptionTarget
+    condition is added in the process), after restart it runs to successful completion.
+    Steps: 1. Select a node to run all Job''s pods to ensure the host marker file
```
Comment (dejanzele) [inline test/conformance/testdata/conformance.yaml:1069]:
```suggestion
    Steps: 1. Select a node to run all Job's pods to ensure the host marker file
```

**Comment by mimowo** [comment]
Rebased after https://github.com/kubernetes/kubernetes/pull/125461 is merged

**Comment by mimowo** [comment]
/test pull-kubernetes-verify
/test pull-kubernetes-integration 
Testing infra overloaded

**Comment by mimowo** [comment]
The code is rebased after https://github.com/kubernetes/kubernetes/pull/125461 and https://github.com/kubernetes/kubernetes/pull/125442 are merged. 
@dejanzele @alculquicondor @atiratree @soltysh PTAL.

**Comment by alculquicondor** [comment]
The following is missing:

> any necessary information in the description to verify that the test meets conformance test requirements, such as links to reports or dashboards that prove lack of flakiness

**Comment by mimowo** [comment]
> The following is missing:

Thanks for pointing this out. Updated the PR description in the "Special notes for your reviewer" section

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
sig apps bits

**Comment by mimowo** [comment]
/assign @smarterclayton 
For conformance approval

**Comment by dims** [comment]
@mimowo can you please take a look at the flakes in the [last two weeks](https://storage.googleapis.com/k8s-triage/index.html?test=(Job%20should%20allow%20to%20use%20a%20pod%20failure%20policy%20to%20ignore%7CJob%20should%20allow%20to%20use%20a%20pod%20failure%20policy%20to%20ignore%7CJob%20should%20allow%20to%20use%20the%20pod%20failure%20policy%20on%20exit%7CJob%20should%20allow%20to%20use%20the%20pod%20failure%20policy%20to%20not%7Cvalidates%20pod%20disruption%20condition%20is%20added%20to%20the%20preempted%20pod))?

**Comment by dims** [comment]
> The suggested path forward

+1 to the 2 suggestions @mimowo

**Comment by soltysh** [comment]
> The suggested path forward:

Sounds perfect.

**Comment by mimowo** [comment]
Great, I have updated this PR accordingly and will follow up in https://github.com/kubernetes/kubernetes/pull/126169. PTAL

**Pull request merged**